plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -115,6 +109,60 @@ void startDevServer(List<File> runnables, MessageConsoleStream stream) {
     }
   }
 
+  /**
+   * Starts the development server in debug mode.
+   *
+   * @param runnables the path to directories that contain configuration files like appengine-web.xml
+   * @param stream the stream to send development server process output to
+   * @param debugPort the port to attach a debugger to if launch is in debug mode
+   */
+  void startDebugDevServer(List<File> runnables, MessageConsoleStream stream, int debugPort) {
+    setServerState(IServer.STATE_STARTING);
+
+    // Create dev app server instance
+    initializeDevServer(stream);
+
+    // Create run configuration
+    DefaultRunConfiguration devServerRunConfiguration = new DefaultRunConfiguration();
+    devServerRunConfiguration.setAppYamls(runnables);
+
+    List<String> jvmFlags = new ArrayList<String>();
+    // FIXME: workaround bug when running on a Java8 JVM
+    // https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/issues/181
+    jvmFlags.add("-Dappengine.user.timezone=UTC");
+
+    if (debugPort <= 0 || debugPort > 65535) {
+      throw new IllegalArgumentException("Debug port is set to " + debugPort
+                                      + ", should be between 1-65535");
+    }
+    jvmFlags.add("-Xdebug");
+    jvmFlags.add("-Xrunjdwp:transport=dt_socket,server=n,suspend=y,quiet=y,address=" + debugPort);
+    devServerRunConfiguration.setJvmFlags(jvmFlags);
+
+    // Run server
+    try {
+      devServer.run(devServerRunConfiguration);
+    } catch (AppEngineException ex) {
+      Activator.logError("Error starting server: " + ex.getMessage());
+      stop(true);
+    }
+  }
+
+  private void initializeDevServer(MessageConsoleStream stream) {
+    LocalAppEngineOutputLineListener outputListener =
+        new LocalAppEngineOutputLineListener(stream);
+
+    CloudSdk cloudSdk = new CloudSdk.Builder()
```

Comment:
Use `com.google.cloud.tools.eclipse.sdk`'s `CloudSdkProvider#createBuilder()` to create the builder.